### PR TITLE
Hyperpod runtime precheck script added

### PR DESCRIPTION
**Pull request by DarkSector**
*Issue #, if available:*

*Description of changes:*
This script lets users validate the runtime after cluster creation. It also lets users test their code against unsupported options

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by verdimrc — COMMENTED** [review]
could you document where should this script run? Thanks.

**Comment by sean-smith** [comment]
@DarkSector Let's do this - can you add this in a sub directory of `5.sagemaker-hyperpod` with a README.md that describes all the checks.

**Comment by sean-smith** [comment]
@DarkSector Ping on the README ask.

File: 1.architectures/5.sagemaker-hyperpod/README.md [hunk 1]
```diff
@@ -311,7 +312,39 @@ dev*         up   infinite      4   idle ip-10-1-4-190,ip-10-1-5-138,ip-10-1-18-
...
+| nvidia_cli_installed           | Nvidia Container CLI is installed via docker life cycle scripts. It's unlikely this will be an issue.                       | Go to [this page](https://catalog.workshops.aws/sagemaker-hyperpod/en-US/03-megatron-lm/01-pre-process) and look for the command that runs the nvidia-container-cli installation.<br /> Create a script from those steps and either use sbatch or srun to execute across all compute nodes<br>You can also use this same script to check for unsupported operations in your training launch script |
+
+
+You can also run validation on the scripts you wish to run. This ensures you’re not using unsupported operations in the script
+
+```
+python3 hyperpod-precheck.py -f awsome-distributed-training/3.test_cases/1.megatron-lm/2.distributed-training.sbatch
```
Comment (sean-smith) [inline 1.architectures/5.sagemaker-hyperpod/README.md:343]: use a relative link not full path i.e. 

```bash
python3 hyperpod-precheck.py -f 2.distributed-training.sbatch
```

File: 1.architectures/5.sagemaker-hyperpod/README.md [hunk 2]
```diff
@@ -311,7 +312,39 @@ dev*         up   infinite      4   idle ip-10-1-4-190,ip-10-1-5-138,ip-10-1-18-
 
 You'll also find your FSx for Lustre volume mounted at `/fsx`.
 
-### 3.5 Patching your HyperPod cluster
+### 3.5 Runtime validation before running workloads
+
+We've included a runtime validation script`hyperpod-precheck.py` which lets you check the runtime before running any production workloads.
```
Comment (sean-smith) [inline 1.architectures/5.sagemaker-hyperpod/README.md:317]: Space between "script`hyperpod-precheck.py`"

File: 1.architectures/5.sagemaker-hyperpod/README.md (hunk 1 above)
Comment (DarkSector) [reply 1.architectures/5.sagemaker-hyperpod/README.md:343]: Oh this script would be available under `1.architectures/5.sagemaker-hyperpod` so wouldn't relative path be `../../3.test_cases/1.megatron-lm`? Or should it be in the directory with source?

File: 1.architectures/5.sagemaker-hyperpod/README.md (hunk 2 above)
Comment (DarkSector) [reply 1.architectures/5.sagemaker-hyperpod/README.md]: Fixed

**Review by sean-smith — APPROVED** [review]

**Pull request merged**